tests/test_table.py

```diff
@@ -1298,6 +1307,111 @@ def test_add_group_metadata_w_existing_metadata(self):
+                                ['Bacteria', 'Bacteroidetes']],
+                               index=['O1', 'O2'],
+                               columns=['taxonomy_0', 'taxonomy_1'])
+        obs_samp = example_table.metadata_to_dataframe(axis='sample')
+        obs_obs = example_table.metadata_to_dataframe(axis='observation')
+        pdt.assert_frame_equal(obs_samp, exp_samp)
+        pdt.assert_frame_equal(obs_obs, exp_obs)
+
+    def test_metadata_to_dataframe_uneven_list_metadata(self):
+        tab = Table(np.array([[1,2],[3,4]]), ['a', 'b'], ['c', 'd'],
+                    [{'taxonomy': ['k__foo', 'p__bar']},
+                     {'taxonomy': ['k__foo']}])
+        exp_obs = pd.DataFrame([['k__foo', 'p__bar'],
+                                ['k__foo', None]],
+                               index=['a', 'b'],
+                               columns=['taxonomy_0', 'taxonomy_1'])
+        obs_obs = tab.metadata_to_dataframe(axis='observation')
+        pdt.assert_frame_equal(obs_obs, exp_obs)
+
+    def test_metadata_to_dataframe_badaxis(self):
+        with self.assertRaises(UnknownAxisError):
+            example_table.metadata_to_dataframe(axis='foo')
+
+    def test_metadata_to_dataframe_nomd(self):
+        tab = Table(np.array([[1,2], [3,4]]), ['a', 'b'], ['c', 'd'],
+                    [{'foo': 1}, {'foo': 2}])
+        with self.assertRaises(KeyError):
+            tab.metadata_to_dataframe('sample')
+
+    def test_del_metadata_full(self):
+        obs_ids = [1, 2, 3]
+        obs_md = [{'taxonomy': ['A', 'B'], 'other': 'h1'},
+                  {'taxonomy': ['B', 'C'], 'other': 'h2'},
+                  {'taxonomy': ['E', 'D', 'F'], 'other': 'h3'}]
+        samp_ids = [4, 5, 6, 7]
+        samp_md = [{'foo': 1}, {'foo': 2}, {'foo': 3}, {'foo': 4}]
+        d = np.array([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]])
+        t = Table(d, obs_ids, samp_ids, observation_metadata=obs_md,
+                  sample_metadata=samp_md)
+        exp = Table(d, obs_ids, samp_ids)
+        obs = t.del_metadata(axis='whole')
+        self.assertEqual(obs, exp)
+
+    def test_del_metadata_partial(self):
+        obs_ids = [1, 2, 3]
+        obs_md = [{'taxonomy': ['A', 'B'], 'other': 'h1'},
+                  {'taxonomy': ['B', 'C'], 'other': 'h2'},
+                  {'taxonomy': ['E', 'D', 'F'], 'other': 'h3'}]
+        samp_ids = [4, 5, 6, 7]
+        samp_md = [{'d': 0}, {'e': 0}, {'f': 0}, {'g': 0}]
+        d = np.array([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]])
+        t = Table(d, obs_ids, samp_ids, observation_metadata=obs_md,
+                  sample_metadata=samp_md)
+
+        exp_md = [d.copy() for d in obs_md]
+        del exp_md[0]['taxonomy']
+        del exp_md[1]['taxonomy']
+        del exp_md[2]['taxonomy']
+
+        exp = Table(d, obs_ids, samp_ids, observation_metadata=exp_md,
+                    sample_metadata=samp_md)
+        obs = t.del_metadata(keys=['taxonomy'], axis='observation')
```

Comment:
Can you add tests for `keys` that are an empty list, and a list with more than one element?